File: compare.py
import json

with open("C:/Users/MuraliM/.gemini/antigravity/brain/cb56279e-c6a0-4f88-ba0d-18b25d092aba/.system_generated/steps/234/output.txt", "r", encoding="utf-8") as f:
    schema1 = json.load(f)["tables"]
with open("C:/Users/MuraliM/.gemini/antigravity/brain/cb56279e-c6a0-4f88-ba0d-18b25d092aba/.system_generated/steps/235/output.txt", "r", encoding="utf-8") as f:
    schema2 = json.load(f)["tables"]

d1 = {t["name"]: t for t in schema1}
d2 = {t["name"]: t for t in schema2}

print("=== TABLE DIFFERENCES ===")
diff1 = set(d1.keys()) - set(d2.keys())
print("Tables in DTEAA not in muralineo:", diff1 if diff1 else "None")
diff2 = set(d2.keys()) - set(d1.keys())
print("Tables in muralineo not in DTEAA:", diff2 if diff2 else "None")

print("\n=== COLUMN DIFFERENCES IN COMMON TABLES ===")
for t in set(d1.keys()) & set(d2.keys()):
    c1 = {c["name"]: c for c in d1[t]["columns"]}
    c2 = {c["name"]: c for c in d2[t]["columns"]}
    missing_in_2 = set(c1.keys()) - set(c2.keys())
    missing_in_1 = set(c2.keys()) - set(c1.keys())
    if missing_in_2 or missing_in_1:
        print(f"Table {t}:")
        if missing_in_2: print(f"  Missing in muralineo: {missing_in_2}")
        if missing_in_1: print(f"  Missing in DTEAA: {missing_in_1}")
        
    for col in set(c1.keys()) & set(c2.keys()):
        if c1[col].get("data_type") != c2[col].get("data_type"):
            print(f"  Table {t} Column {col} Type mismatch: DTEAA={c1[col].get('data_type')} vs muralineo={c2[col].get('data_type')}")

print("\n=== FOREIGN KEY DIFFERENCES IN COMMON TABLES ===")
for t in set(d1.keys()) & set(d2.keys()):
    f1 = {f["name"] for f in d1[t].get("foreign_key_constraints", [])}
    f2 = {f["name"] for f in d2[t].get("foreign_key_constraints", [])}
    if f1 != f2:
        print(f"Table {t}:")
        if f1 - f2: print(f"  FKs only in DTEAA: {f1 - f2}")
        if f2 - f1: print(f"  FKs only in muralineo: {f2 - f1}")

File: compare.js
const fs = require('fs');

const schema1Raw = fs.readFileSync('C:/Users/MuraliM/.gemini/antigravity/brain/cb56279e-c6a0-4f88-ba0d-18b25d092aba/.system_generated/steps/234/output.txt', 'utf8');
const schema2Raw = fs.readFileSync('C:/Users/MuraliM/.gemini/antigravity/brain/cb56279e-c6a0-4f88-ba0d-18b25d092aba/.system_generated/steps/235/output.txt', 'utf8');

const schema1 = JSON.parse(schema1Raw).tables;
const schema2 = JSON.parse(schema2Raw).tables;

const d1 = Object.fromEntries(schema1.map(t => [t.name, t]));
const d2 = Object.fromEntries(schema2.map(t => [t.name, t]));

const keys1 = new Set(Object.keys(d1));
const keys2 = new Set(Object.keys(d2));

console.log("=== TABLE DIFFERENCES ===");
const diff1 = [...keys1].filter(x => !keys2.has(x));
console.log("Tables in DTEAA not in muralineo:", diff1.length ? diff1 : "None");

const diff2 = [...keys2].filter(x => !keys1.has(x));
console.log("Tables in muralineo not in DTEAA:", diff2.length ? diff2 : "None");

console.log("\n=== COLUMN DIFFERENCES IN COMMON TABLES ===");
const commonTables = [...keys1].filter(x => keys2.has(x));

for (const t of commonTables) {
    const c1 = Object.fromEntries(d1[t].columns.map(c => [c.name, c]));
    const c2 = Object.fromEntries(d2[t].columns.map(c => [c.name, c]));
    
    const ckeys1 = new Set(Object.keys(c1));
    const ckeys2 = new Set(Object.keys(c2));
    
    const missingIn2 = [...ckeys1].filter(x => !ckeys2.has(x));
    const missingIn1 = [...ckeys2].filter(x => !ckeys1.has(x));
    
    if (missingIn2.length || missingIn1.length) {
        console.log(`Table ${t}:`);
        if (missingIn2.length) console.log(`  Missing in muralineo: ${missingIn2}`);
        if (missingIn1.length) console.log(`  Missing in DTEAA: ${missingIn1}`);
    }
    
    const commonCols = [...ckeys1].filter(x => ckeys2.has(x));
    for (const col of commonCols) {
        if (c1[col].data_type !== c2[col].data_type) {
            console.log(`  Table ${t} Column ${col} Type mismatch: DTEAA=${c1[col].data_type} vs muralineo=${c2[col].data_type}`);
        }
    }
}

console.log("\n=== FOREIGN KEY DIFFERENCES IN COMMON TABLES ===");
for (const t of commonTables) {
    const f1 = new Set((d1[t].foreign_key_constraints || []).map(f => f.name));
    const f2 = new Set((d2[t].foreign_key_constraints || []).map(f => f.name));
    
    const onlyIn1 = [...f1].filter(x => !f2.has(x));
    const onlyIn2 = [...f2].filter(x => !f1.has(x));
    
    if (onlyIn1.length || onlyIn2.length) {
        console.log(`Table ${t}:`);
        if (onlyIn1.length) console.log(`  FKs only in DTEAA: ${onlyIn1}`);
        if (onlyIn2.length) console.log(`  FKs only in muralineo: ${onlyIn2}`);
    }
}
